pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -157,14 +177,31 @@ func (nm *NodeManager) DiscoverNode(nodeID string, searchBy cm.FindVM) error {
 		return err
 	}
 
+	vcInstance := nm.connectionManager.VsphereInstanceMap[vmDI.VcServer]
+
+	ipFamily := []string{vcfg.DefaultIPFamily}
+	if vcInstance != nil {
+		ipFamily = vcInstance.Cfg.IPFamilyPriority
+	} else {
+		klog.Warningf("Unable to find vcInstance for %s. Defaulting to ipv4.", vmDI.VcServer)
+	}
+
+	found := false
 	addrs := []v1.NodeAddress{}
 	for _, v := range oVM.Guest.Net {
 		if v.DeviceConfigId == -1 {
 			klog.V(4).Info("Skipping device because not a vNIC")
 			continue
 		}
-		for _, ip := range v.IpAddress {
-			if net.ParseIP(ip).To4() != nil {
+
+		// Only return a single IP address based on the preference of IPFamily
+		// Must break out of loop in the event of ipv6,ipv4 where the NIC does
+		// contain a valid IPv6 and IPV4 address
+		for _, family := range ipFamily {
+			ips := returnIPsFromSpecificFamily(family, v.IpAddress)
+
+			for _, ip := range ips {
+				klog.V(2).Infof("Adding IP: %s", ip)
```

Comment:
This seems pretty verbose for v=2?

Reply:
I thought it was super helpful to know what address is being used especially when you run in dual stack mode when you can get a mix. I think it's worth the extra line per node in the log file.